lib/model.js

```diff
@@ -683,7 +683,7 @@ class Model {
     }
     this.sequelize = options.sequelize;
 
-    const globalOptions = this.sequelize.options;
+    const globalOptions = _.cloneDeep(this.sequelize.options);
```

Comment:
It appears to me that `globalOptions` is not used besides in L688. if that is the case, I think we could just modify L689 to read
```js
options = Utils.merge({}, globalOptions.define, options)
```